rosbag2/include/rosbag2/sequential_reader.hpp

```diff
@@ -0,0 +1,90 @@
+// Copyright 2018, Bosch Software Innovations GmbH.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+#ifndef ROSBAG2__SEQUENTIAL_READER_HPP_
+#define ROSBAG2__SEQUENTIAL_READER_HPP_
+
+#include <memory>
+#include <string>
+#include <vector>
+
+#include "rosbag2_storage/storage_factory.hpp"
+#include "rosbag2_storage/storage_interfaces/read_only_interface.hpp"
+#include "rosbag2/storage_options.hpp"
+#include "rosbag2/types.hpp"
+#include "rosbag2/visibility_control.hpp"
+
+// This is necessary because of using stl types here. It is completely safe, because
+// a) the member is not accessible from the outside
+// b) there are no inline functions.
+#ifdef _WIN32
+# pragma warning(push)
+# pragma warning(disable:4251)
+#endif
+
+namespace rosbag2
+{
+/**
+ * The SequentialReader allows opening and reading messages of a bag. Messages will be read
+ * sequentially according to timestamp.
+ */
+class ROSBAG2_PUBLIC SequentialReader
+{
+public:
+  virtual ~SequentialReader();
+
+  /**
+   * Open a rosbag for reading messages sequentially (time-ordered). Throws if file could not be
+   * opened. This must be called before any other function is used. The rosbag is
+   * automatically closed on destruction.
+   *
+   * \param options Options to configure the storage
+   */
+  virtual void open(const StorageOptions & options);
```

Comment:
Google style guide uses uppercase function names e.g. ``Open()``... are you keeping the ROS1 style on purpose?

Reply:
Most of the ROS 2 code doesn't use CamelCase for functions or methods. I'm not sure what the original reason for that was, but functions, like variable names are typically all lower-case with underscores (a.k.a. `snake_case`), which is the same style the C++ standard library uses. One reason mentioned in our developer guide (https://github.com/ros2/ros2/wiki/Developer-Guide#c-1) (with respect to variable names not function names, but still) that this is done to have better symmetry between languages like Python and pure C where snake case is more common and CamelCase is reserved for class names.

Reply:
I read the C++ style guideline for ROS2 earlier today and am sad its still "Google style except all these exceptions". I also checked for this breakage of Google rule and as you pointed out, its not documented there yet. I really like the Google style guide but I'll stop there since I haven't really been involved in all the debates with ROS2.

I will also point out that Google style uses underscores when its a pure getter e.g. ``controller_.num_cycles()`` but not when its a real function e.g. ``controller_.RunLoop()``. But that doesn't matter for ROS2 as much since it allows for using public member variables directly (sloppy!).

Reply:
thanks for the explanations!

Reply:
I updated our developer guide with the function naming stuff:

https://github.com/ros2/ros2/wiki/Developer-Guide/_compare/eddd356eaa407b62426e19abb27fa74722e36b2c

If you want to discuss it further let's take it off this pull request. (issue on ros2/ros2 maybe?)